### PR TITLE
Add remote ruleset to woke configuration

### PR DESCRIPTION
**Pull request by mkcomer**
Please check if the PR fulfills these requirements

- [x]  The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
Feature

What is the current behavior? (You can also link to an open issue here)
Link to issue: #1 

What is the new behavior (if this is a feature change)?
This PR adds functionality to pull in remote rules from urls - https.

To test this PR:
go build
go run main.go -c "https://raw.githubusercontent.com/get-woke/woke/main/example.yaml" --debug

To test a 404 URL:
go run main.go -c "https://raw.githubusercontent.com/get-woke/woke/main/example" --debug

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -81,14 +86,77 @@ func (c *Config) ConfigureRules() {
...
+			return err
+		}
+		defer out.Close()
+
+		log.Debug().Int("here", resp.StatusCode).Msg("help")
+		// Write the body to file
+		_, err = io.Copy(out, resp.Body)
+		return err
+	} else {
+		return fmt.Errorf("unable to download remote config from url - http response code: %v", resp.StatusCode)
```
Comment (jeremydelacruz) [inline pkg/config/config.go:156]: Totally agree with returning the error here -- maybe we should also log.Error() the msg/body of the http response here as well? Just thinking it might provide more detailed information in case the remote server sent relevant info in the response body

File: pkg/config/config_test.go [hunk 2]
```diff
@@ -155,6 +157,51 @@ func TestNewConfig(t *testing.T) {
...
+		assert.True(t, boolResponse)
+	})
+
+	t.Run("invalid-url", func(t *testing.T) {
+		boolResponse := isValidURL("Users/Document/test.yaml")
+		assert.False(t, boolResponse)
+	})
+
+	t.Run("invalid-url", func(t *testing.T) {
+		boolResponse := isValidURL("/Users/Document/test.yaml")
```
Comment (jeremydelacruz) [inline pkg/config/config_test.go:186]: Out of curiosity, do you know if something like `C:\user\test.yaml` or `C:\directory.com\test.yaml` returns valid? Unsure how the URL validation works under the hood and just thinking of some weird edge cases

File: pkg/config/config.go [hunk 3]
```diff
@@ -81,14 +86,77 @@ func (c *Config) ConfigureRules() {
...
+	if err != nil || u.Scheme == "" || u.Host == "" {
+		return false
+	}
+	log.Debug().Str("remoteConfig", toTest).Msg("Valid URL for remote config.")
+	return true
+}
+
+// downloads file from url to set filepath
+func DownloadFile(filepath string, url string) error {
+	// Removing gosec lint warning - as we have to pass in user specified url for remote config
```
Comment (jeremydelacruz) [inline pkg/config/config.go:128]: This comment looks like it's from when we were doing `http.Get()` to document that we were suppressing the linting warnings -- is it still needed?

**Review by jeremydelacruz — APPROVED** [review]
Just a couple of questions and suggestions, but looks good to me!

File: pkg/config/config.go (hunk 3 above)
Comment (mkcomer) [reply pkg/config/config.go:128]: Good call - forgot to remove the warning !

File: pkg/config/config_test.go (hunk 2 above)
Comment (mkcomer) [reply pkg/config/config_test.go:186]: it returns invalid  - adding in some more unit tests

File: pkg/config/config.go (hunk 1 above)
Comment (mkcomer) [reply pkg/config/config.go]: Added in the response body to the error message!

File: pkg/config/config.go [hunk 4]
```diff
@@ -81,14 +86,74 @@ func (c *Config) ConfigureRules() {
...
+		downloadedFile := "downloadedRules.yaml"
+		err := DownloadFile(downloadedFile, filename)
+		if err != nil {
+			return c, err
+		}
+		filename = downloadedFile
+		log.Debug().Str("filename", filename).Msg("Saved remote config to local file.")
+	}
+
+	// TO DO = Need to add more error handling on reading/validating yamlFile file - issue # 16
```
Comment (KSLHacks) [inline pkg/config/config.go:102]: I think this is fine keeping the TODO item here for visibility, but keep in mind referencing issue #16 upstream won't make sense.

**Review by KSLHacks — COMMENTED** [review]
PR looks good! I suggest adding a section into the README docs to explain the functionality and how to use the CLI command.

Other than documentation code is readable.

File: pkg/config/config.go (hunk 4 above)
Comment (mkcomer) [reply pkg/config/config.go:102]: good call - i may just remove for now

**Review by KSLHacks — APPROVED** [review]
